src/api.js

```diff
@@ -57,15 +57,26 @@ class Api {
 	async query( sql, params ) {
 		const _sql = await sql;
 		return this.withConnection( conn => {
-			const data = [];
+			const sets = [];
+			let data;
 			return new Promise( ( resolve, reject ) => {
 				const request = new Request( _sql, err => {
 					if ( err ) {
 						return reject( err );
 					}
+					if ( sets.length ) {
```

Comment:
How could `sets` ever be empty here? If the query return no result sets?

Reply:
That's where this gets weird. The event order for a single set is:
1 columnMetadata event
n row events
request callback

The event order for multiple is:
1 columnMetadata event
n row events
1 columnMetadata event
n row events
...
request callback

`sets` will be empty if the request callback gets called after only a single `columnMetadata` event.

Reply:
Ah, I see it now. I was overlooking how the first time through, `data` had not been assigned yet.